scripts/logs/logs_analysis/context.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import common.models.log_reading as log_types
+
+from bisect import bisect_left
+from common.models.visitor import LogsVisitor
+
+
+class AnalysisContext:
+    def __init__(self, all_logs: list[log_types.LogReading]) -> None:
+        self.all_logs = all_logs
+
+        self.time_logs = []
+        self.gps_logs = []
+        self.baro_logs = []
+        self.imu_logs = []
+
+        filler = AnalysisContext.ContextFiller(self)
+        for log in all_logs:
+            log.accept(filler)
+
+        self.missed_logs_cnt = self.__calculate_missed_logs_cnt()
+
+    def get_index_from_all_logs(self, log: log_types.LogReading) -> int:
```

Comment:
Name shortening idea:
```suggestion
    def get_indices(self, log: log_types.LogReading) -> int:
```

please add a docstring what it does